src/iconid.rs

```diff
@@ -202,6 +231,102 @@ fn resolve_ligature_internal(
     Ok(None)
 }
 
+/// Returns the ligatures of the given icon font.
+/// Some assumptions are made:
+/// each ligature must have a PUA codepoint assigned in cmap, if non-PUA is assigned, the ligature will be ignored.
+/// each ligature component must have a valid non-PUA codepoint entry in cmap.
+/// glyphs are allowed to be assigned to multiple codepoints.
+/// ligatures are allowed to point to the same glyph.
+/// a glyph with a PUA and non-PUA codepoint is considered as single charcter ligature and will be returned in the result.
+///
```

Comment:
I feel like the code is more complicated than it should be. Some of it looks duplicative of iconid.rs resolve_ligature_iternal, they should share.

Reply:
I would think you could:

Make a set of glyph ids that are icon glyph ids by taking the value side of any cmap entry where the codepoint is a PUA.

Collect the names of single-char identifiers by taking the character from any cmap entry that targets an icon gid where the codepoint is not a PUA.

Collect the names of icons accessed by ligatures by iterating ligatures - perhaps we should have a ligature iterator? - that target an icon glyph id and then reverse cmapping the glyph ids to form a string.

If you make a ligature iterator you can share it with icon id, perhaps also adding a trait implemented on FontRef so you can say things like `font.ligatures()`.

Similarly, you could go so far as to make a trait for icon name enumeration so you can say things like `font.icon_names()`.

Reply:
(this is now done)